cluster-autoscaler/dynamic_reconfiguration.go

```diff
@@ -0,0 +1,48 @@
+package main
+
+import (
+	"fmt"
+	"github.com/golang/glog"
+	"k8s.io/contrib/cluster-autoscaler/cloudprovider/builder"
+	"k8s.io/contrib/cluster-autoscaler/dynamic"
+)
+
+// DynamicReconfiguration dynamically reconfigure cluster-autoscaler with node group specs read from a configmap
+type DynamicReconfiguration struct {
+	configMapName        string
+	autoscalingContext   *AutoscalingContext
+	configFetcher        *dynamic.ConfigFetcher
+	cloudProviderBuilder builder.CloudProviderBuilder
+}
+
+// NewDynamicReconfiguration builds a new dynamic reconfiguration object
+func NewDynamicReconfiguration(configMapName string, autoscalingContext *AutoscalingContext, configFetcher *dynamic.ConfigFetcher, cloudProviderBuilder builder.CloudProviderBuilder) DynamicReconfiguration {
+	return DynamicReconfiguration{
+		configMapName:        configMapName,
+		autoscalingContext:   autoscalingContext,
+		configFetcher:        configFetcher,
+		cloudProviderBuilder: cloudProviderBuilder,
+	}
+}
+
+// Run dynamic reconfiguration once
+func (r DynamicReconfiguration) Run() error {
```

Comment:
Probably no? because what's we're updating is not `DynamicConfiguration` but rather `AutoscalingContext`.